web/js/codeworld_shared.js

```diff
@@ -449,11 +457,18 @@ function substitutionCost(a, b, fixedLen) {
     }
   }
 
-  return d[b.length][a.length] + scale(fixedLen) * (a.length - b.length);
+  return (
+    d[b.length][a.length] +
+    scale(fixedLen) * (a.length - b.length) +
+    (isTermReplaced ? redirectPenalty : 0)
+  );
 }
 
 // Hints and hover tooltips
-function registerStandardHints(successFunc) {
+async function registerStandardHints(successFunc) {
+  const blob = await fetch('./replacement_terms.json');
```

Comment:
Why not do something like this:

```
let replacementTerms = { definingModule: "", mapping: {} };
fetch('./replacement_terms.json').then((blob) => {
  replacementTerms = await blob.json();
});
```

May as well avoid the extra startup delay.

Reply:
What do you have in mind as a replacement?

Reply:
Isn't `async/await` just a syntactic sugar over promises?

Reply:
My understanding is that if you `await` at the beginning of `registerStandardHints`, then the rest of this function will not run until the fetch is complete.  That means we won't have any autocomplete at all, won't even start loading `codeworld-base.txt`, etc., until this fetch is complete.  That's not really needed.  It's okay to run the rest of this function without waiting on that network round trip.

Reply:
Yes, you are right, code execution literally pauses at the line containing the `await` keyword and continues only once the async operations is complete. There is no such issue with promises. Turns out I had a misconception about the way `await` works ... Now it's cleared up. Will update the code.